web_src/js/features/admin/common.js

```diff
@@ -5,136 +5,165 @@ import {POST} from '../../modules/fetch.js';
-          $('#password').attr('required', 'required');
+        if (this.getAttribute('data-password') === 'required') {
+          document.getElementById('password')?.setAttribute('required', 'required');
         }
       } else {
-        if ($('.admin.edit.user').length > 0) {
-          $('#user_name').attr('disabled', 'disabled');
+        if (document.querySelector('.admin.edit.user')) {
+          document.getElementById('user_name')?.setAttribute('disabled', 'disabled');
         }
-        $('#login_name').attr('required', 'required');
+        document.getElementById('login_name')?.setAttribute('required', 'required');
         showElem('.non-local');
         hideElem('.local');
-        $('#login_name').trigger('focus');
+        document.getElementById('login_name')?.focus();
 
-        $('#password').removeAttr('required');
+        document.getElementById('password')?.removeAttribute('required');
       }
     });
   }
 
-  function onSecurityProtocolChange() {
-    if ($('#security_protocol').val() > 0) {
-      showElem('.has-tls');
-    } else {
-      hideElem('.has-tls');
-    }
-  }
-
   function onUsePagedSearchChange() {
+    const searchPageSizeElements = document.querySelectorAll('.search-page-size');
     if (document.getElementById('use_paged_search').checked) {
       showElem('.search-page-size');
-      $('.search-page-size').find('input').attr('required', 'required');
+      for (const el of searchPageSizeElements) {
+        el.querySelector('input')?.setAttribute('required', 'required');
+      }
     } else {
       hideElem('.search-page-size');
-      $('.search-page-size').find('input').removeAttr('required');
+      for (const el of searchPageSizeElements) {
+        el.querySelector('input')?.removeAttribute('required');
+      }
     }
   }
 
   function onOAuth2Change(applyDefaultValues) {
     hideElem('.open_id_connect_auto_discovery_url, .oauth2_use_custom_url');
-    $('.open_id_connect_auto_discovery_url input[required]').removeAttr('required');
+    for (const input of document.querySelectorAll('.open_id_connect_auto_discovery_url input[required]')) {
+      input.removeAttribute('required');
+    }
 
-    const provider = $('#oauth2_provider').val();
+    const provider = document.getElementById('oauth2_provider')?.value;
     switch (provider) {
       case 'openidConnect':
-        $('.open_id_connect_auto_discovery_url input').attr('required', 'required');
+        for (const input of document.querySelectorAll('.open_id_connect_auto_discovery_url input')) {
+          input.setAttribute('required', 'required');
+        }
         showElem('.open_id_connect_auto_discovery_url');
         break;
       default:
-        if ($(`#${provider}_customURLSettings`).data('required')) {
-          $('#oauth2_use_custom_url').attr('checked', 'checked');
+        if (document.getElementById(`#${provider}_customURLSettings`)?.getAttribute('data-required')) {
```

Comment:
This change is wrong (#31244)

I would suggest to avoid using `getElementById` but only use `queryXxx`

Reply:
Good catch. I think `getElementById` may have been used for performance, but as I now see it the difference is neglible: https://old.reddit.com/r/learnjavascript/comments/i0f5o8/performance_of_getelementbyid_vs_queryselector/

So I agree to refactor all `getElementById` to `querySelector`.

Reply:
And one more thing, we can't just do simple search&replace to refactor the legacy code. Actually there is not only one regression. 


There are 2 more regressions here.

* One is `document.getElementById(`oauth2_${custom}`).value = document.getElementById(`${provider}_${custom}`).value;` because the element may not exists.
* Another one is `if (el.getAttribute())` is also wrong (seel below)

Every line of refactored & changed code should be fully tested.

```
<a data-x="false">aaa</a>

const v1 = $('a')[0].getAttribute('data-x');
const v2 = $('a').data('x');
console.log(v1 ? 'yes' : 'no');
console.log(v2 ? 'yes' : 'no');
```

Reply:
One more thought: if a page is quite complex and difficult to maintain (for example .... this "auth edit" page), it's better to refactor and rewrite it, instead of replacing legacy function calls. A fully rewritten page could be clearer and more correct than replacing the functions in the legacy code.

Reply:
Yes, we need to be aware of `.data`'s "Every attempt is made to convert the attribute's string value to a JavaScript value" behaviour. It already came up.

Reply:
PS: I'm pretty sure `document.querySelector('a').getAttribute` wouldn't fly in Typescript because typescript knows the querySelector can return null so would raise a type error.

Reply:
> I would suggest to avoid using getElementById but only use queryXxx

We could enable https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-query-selector.md which will be around 180 occurences currently, but I'm not sure if it's really worth it.